src/materials/ComputeNEMLCPOutput.C

```diff
@@ -0,0 +1,132 @@
+#include "ComputeNEMLCPOutput.h"
+
+registerMooseObject("DeerApp", ComputeNEMLCPOutput);
+
+template <>
+InputParameters
+validParams<ComputeNEMLCPOutput>()
+{
+  InputParameters params = validParams<ComputeNEMLStressUpdate>();
+  params.addParam<bool>("add_cp_output", false, "Give CP output");
+  params.addParam<bool>("crystal_plasticity", false, "Use crystal plasticity material models");
+  params.addParam<UserObjectName>("euler_angle_provider","dummy"
+                                        "Name of Euelr angle provider user object");
+  params.addParam<unsigned int>("grain_id", 0,"ID of the grain for this material");
+  return params;
+}
+
+ComputeNEMLCPOutput::ComputeNEMLCPOutput(const InputParameters & parameters)
+   : ComputeNEMLStressUpdate(parameters),
+    _orientation_q(declareProperty<std::vector<Real>>("orientation_q")),
+    _cpOut(getParam<bool>("add_cp_output")),
+    _cp(getParam<bool>("crystal_plasticity")),
+    _euler(parameters.isParamSetByUser("euler_angle_provider") ? &getUserObject<EulerAngleProvider>("euler_angle_provider") : nullptr), // May be making it a required parameter
+    _grain(getParam<unsigned int>("grain_id"))
+{
+  _cpmodel = static_cast<neml::SingleCrystalModel *>(_model.get());
+  if((_cp == false) ){
+    mooseWarning("crystal plasticity flag is not set but still using crystal plasticity model");
+   }
+
+  if((_cp == true) && (!parameters.isParamSetByUser("grain_id"))){
+    mooseWarning("grain id's not provided, block id will be used for the cp");
+    _given = 0;
+   }
+
+  if((_cp == true) && (_euler == nullptr) ) {
+    mooseWarning("no euler angle file is given for a CP model default orientation will be used");
+   }
+ }
+
+// Assigning Euler angles from file
+void
+ComputeNEMLCPOutput::initQpStatefulProperties()
+{
```

Comment:
Do not duplicate code from ComputeNEMLStressUpdate, just call the superclass method.